Dolphin recompilation failure: libdolphinprivate.so.5 not found

`dolphin` @ `v19.12.3`, `make -j2`, recompiling occasionally fails with

```
...
[ 99%] Linking CXX executable ../../bin/kstandarditemmodeltest
[ 99%] Automatic MOC for target dolphin
[ 99%] Built target kstandarditemmodeltest
[ 99%] Built target dolphin_autogen
Scanning dependencies of target dolphin
[ 99%] Building CXX object src/CMakeFiles/dolphin.dir/dolphin_dummy.cpp.o
[ 99%] Building CXX object src/CMakeFiles/dolphin.dir/dolphin_autogen/mocs_compilation.cpp.o
[100%] Linking CXX executable ../bin/dolphin
/usr/bin/ld: warning: libdolphinprivate.so.5, needed by ../bin/libkdeinit5_dolphin.so, not found (try using -rpath or -rpath-link)
/usr/bin/ld: ../bin/libkdeinit5_dolphin.so: undefined reference to `KItemListView::dragMoveEvent(QGraphicsSceneDragDropEvent*)'
/usr/bin/ld: ../bin/libkdeinit5_dolphin.so: undefined reference to `KItemListView::createDragPixmap(KItemSet const&) const'
/usr/bin/ld: ../bin/libkdeinit5_dolphin.so: undefined reference to `DolphinView::itemCountChanged()'
... (about 300 more undefined references)
collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/dolphin.dir/build.make:99: bin/dolphin] Error 1
make[1]: *** [CMakeFiles/Makefile2:1154: src/CMakeFiles/dolphin.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
```

--- a/src/firebuild/firebuild.cc
+++ b/src/firebuild/firebuild.cc
@@ -474,6 +474,7 @@ void proc_ic_msg(uint32_t ack_num,
              tag == FBB_TAG_dup3 ||
              tag == FBB_TAG_dup ||
              tag == FBB_TAG_rename ||
+             tag == FBB_TAG_symlink ||
              tag == FBB_TAG_fcntl ||
              tag == FBB_TAG_ioctl ||
              tag == FBB_TAG_chdir ||
@@ -604,6 +605,8 @@ void proc_ic_msg(uint32_t ack_num,
         ::firebuild::ProcessPBAdaptor::msg(proc, reinterpret_cast<const FBB_dup *>(fbb_buf));
       } else if (tag == FBB_TAG_rename) {
         ::firebuild::ProcessPBAdaptor::msg(proc, reinterpret_cast<const FBB_rename *>(fbb_buf));
+      } else if (tag == FBB_TAG_symlink) {
+        ::firebuild::ProcessPBAdaptor::msg(proc, reinterpret_cast<const FBB_symlink *>(fbb_buf));
       } else if (tag == FBB_TAG_fcntl) {
         ::firebuild::ProcessPBAdaptor::msg(proc, reinterpret_cast<const FBB_fcntl *>(fbb_buf));
       } else if (tag == FBB_TAG_ioctl) {

--- a/src/firebuild/process.cc
+++ b/src/firebuild/process.cc
@@ -249,6 +249,17 @@ int Process::handle_rename(const std::string &old_ar_name, const std::string &ne
   return 0;
 }
 
+int Process::handle_symlink(const std::string &old_ar_name, const std::string &new_ar_name,
+                            const int error) {
+  if (!error) {
+    disable_shortcutting_bubble_up("Process created a symlink (" +
+                                   pretty_print_string(new_ar_name) + " -> " +
+                                   pretty_print_string(old_ar_name) + ")");
+    return -1;
+  }
+  return 0;
+}
+
 int Process::handle_clear_cloexec(const int fd) {
   if (!get_fd(fd)) {
     disable_shortcutting_bubble_up("Process successfully cleared cloexec on fd (" +

--- a/src/firebuild/process.h
+++ b/src/firebuild/process.h
@@ -215,6 +215,16 @@ class Process {
                     const int error = 0);
 
   /**
+   * Handle symlink()
+   * @param old_ar_name old relative or absolute file name
+   * @param new_ar_name new relative or absolute file name
+   * @param error error code
+   * @return 0 on success, -1 on failure
+   */
+  int handle_symlink(const std::string &old_ar_name, const std::string &new_ar_name,
+                     const int error = 0);
+
+  /**
    * Handle successfully clearing the cloexec bit, via a
    * posix_spawn_file_actions_adddup2() handler with oldfd==newfd.
    * @param fd file descriptor

--- a/src/firebuild/process_proto_adaptor.cc
+++ b/src/firebuild/process_proto_adaptor.cc
@@ -53,6 +53,11 @@ int ProcessPBAdaptor::msg(Process *p, const FBB_rename *r) {
   return p->handle_rename(fbb_rename_get_oldpath(r), fbb_rename_get_newpath(r), error);
 }
 
+int ProcessPBAdaptor::msg(Process *p, const FBB_symlink *s) {
+  const int error = fbb_symlink_get_error_no_with_fallback(s, 0);
+  return p->handle_symlink(fbb_symlink_get_oldpath(s), fbb_symlink_get_newpath(s), error);
+}
+
 int ProcessPBAdaptor::msg(Process *p, const FBB_fcntl *f) {
   const int error = fbb_fcntl_get_error_no_with_fallback(f, 0);
   int arg = fbb_fcntl_get_arg_with_fallback(f, 0);

--- a/src/firebuild/process_proto_adaptor.h
+++ b/src/firebuild/process_proto_adaptor.h
@@ -26,6 +26,7 @@ class ProcessPBAdaptor {
   static int msg(Process *p, const FBB_dup *d);
   static int msg(Process *p, const FBB_dup3 *d);
   static int msg(Process *p, const FBB_rename *r);
+  static int msg(Process *p, const FBB_symlink *s);
   static int msg(Process *p, const FBB_fcntl *f);
   static int msg(Process *p, const FBB_ioctl *i);
   static int msg(Process *p, const FBB_read *r);

